api/controller.py

```diff
@@ -898,15 +899,33 @@ def download_page(self):
         if len(library.cachedmarcfiles) < 1 and exporter:
             body += "<p>" + _("MARC files aren't ready to download yet.") + "</p>"
 
-        if library.cachedmarcfiles:
-            body += "<ul>"
-            for file in library.cachedmarcfiles:
-                if last_update is None or file.representation.mirrored_at > last_update:
-                    last_update = file.representation.mirrored_at
-                body += '<li><a href="%s">%s</a></li>' % (file.representation.mirror_url, file.lane.display_name if file.lane else "All Books")
-            body += "</ul>"
-        if last_update:
-            body += "<p>Last update: %s</p>" % last_update.strftime("%B %-d, %Y")
+        full_files = self._db.query(CachedMARCFile).filter(
+            CachedMARCFile.library==library
+        ).filter(
+            CachedMARCFile.start_time==None
+        )
+
+        updates = self._db.query(CachedMARCFile).filter(
+            CachedMARCFile.library==library
+        ).filter(
+            CachedMARCFile.start_time!=None
+        ).order_by(CachedMARCFile.end_time.desc())
```

Comment:
My reading of the core branch is that end_time will always be present. If that's true, then the original record would show up in both lists.

But I'm reading this as a different design, where we periodically create full MARC record dumps but also have updates between the full dumps.

Can you explain (to me, not in the code) which one we're doing? I see issues with either approach but I don't know which one to focus on.

Either way, I think the "All records" wording is confusing -- Something like "Full recordsets" would work better. "All records" implies that the data is current.

Reply:
end_time will always be present, but start_time is only present for the updates, not the full dumps. This code uses the presence of start_time to determine where to show a file. You're right that we periodically create full MARC record dumps but also have updates between the full dumps.

Reply:
That sounds like a full record dump supercedes all previous delta dumps. In that case I think it would be better to show a single chronological list of the dumps with the deltas in nested lists:

```
<ul>
<li>Full dump 1</li>
<ul>
<li>Update 1A</li>
<li>Update 1B</li>
</ul>
<li>Full dump 2</li>
<ul>
<li>Update 2A</li>
<li>Update 2B</li>
<li>Update 2C</li>
</ul>
</ul>
```

Reply:
I'm not sure if "Full dump" would mean anything to a cataloger but I'll do something like this.